ui/app/components/mount-backend/type-form.js

```diff
@@ -26,11 +25,15 @@ export default class MountBackendTypeForm extends Component {
   @service version;
 
   get secretEngines() {
-    return this.version.isEnterprise ? allEngines() : mountableEngines();
+    return this.version.isEnterprise
```

Comment:
Some comments might help a future dev
```suggestion
// If an enterprise license is present, return all secret engines;
// otherwise, return only the secret engines supported in OSS.
    return this.version.isEnterprise
```